### PR TITLE
Update compat

### PR DESCRIPTION
**Pull request by JeffFessler**
Thanks for accepting the previous PR.  Unfortunately when I added the `[compat]` section I forgot to do a package `update` first, so I ended up pinning things to old versions of the packages, which then causes problems when trying to use this package with others.  This PR simply updates the `[compat]` section to the latest versions of all the relevant packages.
Frankly I don't really understand why upper bounds in compat are required now, so I am not 100% sure that I have done this in the best way here.  The good news at least it seems to pass the tests with the current versions of the included packages.

### CODE REVIEW

**Comment by JeffFessler** [comment]
So this update passes Julia 1.4 but fails tests for earlier versions of Julia.
I guess one or more of the other packages must require Julia 1.4.
My view is that there is little point in supporting old versions of Julia with this update of Wavelets.jl (people using older versions of Julia can use earlier releases of this package) so I have removed the tests of older versions from the CI file.  Maybe there is some way to support both but I don't know how...

**Comment by johnnychen94** [comment]
The test fails because you have `Manifest.toml` checked in, those `*_jll` packages are artifacts and is not compatible to Julia 1.3.

Once `Manifest.toml` is removed, it should be good.

File: Project.toml [hunk 1]
```diff
@@ -18,10 +11,16 @@ Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
+[compat]
+DSP = "0.5.1, 0.6"
+FFTW = "0.2.4, 1.2"
+Reexport = "0.2.0, 0.2"
+SpecialFunctions = "0.7.1, 0.10"
```
Comment (johnnychen94) [inline Project.toml:18]: It's not common to have a gap between two versions:

```suggestion
SpecialFunctions = "0.7.1, 0.8, 0.9, 0.10"
```

File: Project.toml [hunk 2]
```diff
@@ -18,10 +11,16 @@ Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
+[compat]
+DSP = "0.5.1, 0.6"
+FFTW = "0.2.4, 1.2"
+Reexport = "0.2.0, 0.2"
```
Comment (johnnychen94) [inline Project.toml:17]:
```suggestion
Reexport = "0.2"
```

`0.2` is equivalent to `0.2.0` is equivalent to `^0.2.0`.

https://julialang.github.io/Pkg.jl/dev/compatibility/#Version-specifier-format-1

File: Project.toml [hunk 3]
```diff
@@ -18,10 +11,16 @@ Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
+[compat]
+DSP = "0.5.1, 0.6"
+FFTW = "0.2.4, 1.2"
```
Comment (johnnychen94) [inline Project.toml:16]:
```suggestion
FFTW = "0.3, 1"
```

`0.2.4` is a 0.7/1.0 version so we could drop it now.

**Comment by JeffFessler** [comment]
Thanks to @johnnychen94's help, the tests now pass so this is ready for @gummif to review.

**Comment by gummif** [comment]
Thanks

**Pull request merged**